src/components/GLStyleService.js

```diff
@@ -0,0 +1,105 @@
+goog.provide('ga_gl_style_service');
+(function() {
+  var module = angular.module('ga_gl_style_service', []);
+
+  module.provider('gaGLStyle', function() {
+    this.$get = function($http, $q, $window) {
+      var GLStyle = function() {
+        this.styleJSONCache = null;
+
+        this.spriteJSONCache = null;
+
+        this.filters = [];
+
+        this.edits = [];
+
+        this.get = function(styleUrl) {
+          var that = this;
+          var defer = $q.defer();
+          $http.get(styleUrl, { cache: true }).then(function(response) {
+            that.styleJSONCache = response.data;
+            // load sprite
+            var spriteUrl = that.styleJSONCache.sprite + '.json';
+            $http.get(spriteUrl, { cache: true }).then(function(res) {
+              that.spriteJSONCache = res.data;
+              defer.resolve({
+                styleJSON: that.styleJSONCache,
+                spriteJSON: that.spriteJSONCache
+              });
+            }, function(reason) {
+              that.spriteJSON = null;
+              $window.console.error(
+                'Unable to load ' + spriteUrl + ' because ' + reason
+              );
+              // failing to load sprite is ok for now...
+              defer.resolve({
+                styleJSON: that.styleJSONCache,
+                spriteJSON: that.spriteJSONCache
```

Comment:
You can use the next ` then`  to avoid duplicate resolvation, see https://github.com/geoadmin/mf-geoadmin3/blob/mvt/src/components/map/LayersService.js#L798